crates/node/src/commands/run.rs

```diff
@@ -146,14 +178,45 @@ pub fn run_node(mut cli: Cli) -> Result<()> {
 
     let settlement_config: Option<(SettlementLayer, PathBuf)> = match cli.run.settlement {
         Some(SettlementLayer::Ethereum) => {
-            let settlement_conf = cli.run.settlement_conf.expect("clap requires da_conf when settlement is present");
-            if !settlement_conf.exists() {
-                return Err(sc_cli::Error::Input(format!(
-                    "Ethereum config not found at {}",
-                    settlement_conf.display()
-                )));
-            }
-            Some((SettlementLayer::Ethereum, settlement_conf))
+            let settlement_conf = match cli.run.settlement_conf {
+                Some(settlement_conf) => Some(settlement_conf),
+                None => {
+                    let path_base_path = cli.run.base_path()?;
+                    let path_sett_conf_json = path_base_path.path().join("settlement_conf.json");
+                    let path_sett_conf_toml = path_base_path.path().join("settlement_conf.toml");
+                    let (test_path_sett_conf_json,test_path_sett_conf_toml) = match (path_sett_conf_json.exists(),path_sett_conf_toml.exists()) {
+                        (true,true) => {
+                            return Err(sc_cli::Error::Input(format!(
+                                "Ethereum config not found in fn run_node"
+                            )));
+                        },
+                        (true,false) => {
+                            let file = File::open(path_sett_conf_json).unwrap();
+                            let reader = BufReader::new(file);
+                            let res : PathBuf = serde_json::from_reader(reader).unwrap();
+                            (Some(res),None)
+                        },
+                        (false,true) => {
+                            let file = File::open(path_sett_conf_toml).unwrap();
+                            let reader = BufReader::new(file);
+                            let res : PathBuf = serde_json::from_reader(reader).unwrap();
+                            (Some(res),None)
+                        },
+                        _ => {
+                            return Err(sc_cli::Error::Input(format!(
+                                "Ethereum config not found in fn run_node"
+                            )));
+                        }
+                    };
+                    match (test_path_sett_conf_json,test_path_sett_conf_toml) {
+                        (Some(x),None) => Some(x),
+                        (None,Some(x)) => Some(x),
+                        _ => None
+                    }
+                }
```

Comment:
Refactor by creating a generic function that will work for both cases rather than copy pasting